be/src/formats/parquet/file_writer.cpp

```diff
@@ -0,0 +1,435 @@
+// mainly includes value buffer size and repetition buffer size and definition buffer value for each column.
+std::size_t FileWriterBase::_get_current_rg_written_bytes() const {
+    if (_rg_writer == nullptr) {
+        return 0;
+    }
+
+    auto estimated_bytes = std::accumulate(_buffered_values_estimate.begin(), _buffered_values_estimate.end(), 0);
+
+    return _rg_writer->total_bytes_written() + _rg_writer->total_compressed_bytes() + estimated_bytes;
+}
+
+std::size_t FileWriterBase::file_size() const {
+    DCHECK(_outstream != nullptr);
+    return _outstream->Tell().MoveValueUnsafe() + _get_current_rg_written_bytes();
+}
+
+Status FileWriterBase::split_offsets(std::vector<int64_t>& splitOffsets) const {
+    if (_file_metadata == nullptr) {
+        LOG(WARNING) << "file metadata null";
+        return Status::InternalError("Get split offsets while the file metadata is null");
+    }
+    for (int i = 0; i < _file_metadata->num_row_groups(); i++) {
+        auto first_column_meta = _file_metadata->RowGroup(i)->ColumnChunk(0);
+        int64_t dict_page_offset = first_column_meta->dictionary_page_offset();
+        int64_t first_data_page_offset = first_column_meta->data_page_offset();
+        int64_t split_offset = dict_page_offset > 0 && dict_page_offset < first_data_page_offset
+                                       ? dict_page_offset
+                                       : first_data_page_offset;
+        splitOffsets.emplace_back(split_offset);
+    }
+    return Status::OK();
+}
+
+void SyncFileWriter::_flush_row_group() {
+    _rg_writer->Close();
+    _rg_writer = nullptr;
+    std::fill(_buffered_values_estimate.begin(), _buffered_values_estimate.end(), 0);
+}
+
+Status SyncFileWriter::close() {
+    if (_closed) {
+        return Status::OK();
+    }
+
+    _writer->Close();
+    _rg_writer = nullptr;
+    auto st = _outstream->Close();
+    if (st != ::arrow::Status::OK()) {
+        return Status::InternalError("Close file failed!");
+    }
+
+    _closed = true;
+    return Status::OK();
+}
+
+AsyncFileWriter::AsyncFileWriter(std::unique_ptr<WritableFile> writable_file, std::string file_name,
+                                 std::string& file_dir, std::shared_ptr<::parquet::WriterProperties> properties,
+                                 std::shared_ptr<::parquet::schema::GroupNode> schema,
+                                 const std::vector<ExprContext*>& output_expr_ctxs, RuntimeProfile* parent_profile)
+        : FileWriterBase(std::move(writable_file), std::move(properties), std::move(schema), output_expr_ctxs),
+          _file_name(file_name),
+          _file_dir(file_dir),
+          _parent_profile(parent_profile) {
+    _io_timer = ADD_TIMER(_parent_profile, "FileWriterIoTimer");
+}
+
+void AsyncFileWriter::_flush_row_group() {
+    {
+        auto lock = std::unique_lock(_m);
+        _rg_writer_closing = true;
+    }
+    bool ret = ExecEnv::GetInstance()->pipeline_sink_io_pool()->try_offer([&]() {
```

Comment:
It's better to pass `ExecEnv::GetInstance()->pipeline_sink_io_pool()` as a ctor argument instead of let `AsyncFileWriter` call it explicitly.